src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,109 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
 
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            resize();
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size == elementData.length) {
+            resize();
+        }
+        if (index > size) {
+            throw new ArrayIndexOutOfBoundsException(
+                    "Index [" + index + "] is out of size [" + size + "]");
+        }
```

Comment:
please move this check to the very start of the method
we should not make any resize operations unless we are sure that the index is valid

also, we should throw an exception as well if index < 0

Reply:
Done